Topic1/src/Offer.java

```diff
@@ -0,0 +1,62 @@
+/**
+* @author Melina
+* @version 1.0 Composite Pattern
+*/ 
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class Offer implements IOffer{
+	private String name;
+	private double cost;
+	private List<IOffer> offers = new ArrayList<IOffer>();
+	
+	public Offer(String name, double cost){
+		this.name=name;
+		this.cost=cost;
+	}
+
+	public void addOffers(IOffer offer) {
+		this.offers.add(offer);
+	}
+	
+	public void removeOffers(IOffer offer) {
+		this.offers.remove(offer);
+	}
+
+	public double getRealCost() {
+		double price=0;
+		for (IOffer p: offers){
+			 price += p.getCost();
+		}
+		return price;
+	}
+	
+	@Override
+	public String showOffer() {
```

Comment:
Warning: Ideally how the offer is shown should be implemented using a decorator design pattern.